source/common/network/win32_socket_handle_impl.h

```diff
@@ -42,8 +44,25 @@ class Win32SocketHandleImpl : public IoSocketHandleImpl {
                                    RecvMsgOutput& output) override;
   Api::IoCallUint64Result recv(void* buffer, size_t length, int flags) override;
 
+  void initializeFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
+                           Event::FileTriggerType trigger, uint32_t events) override;
+  void enableFileEvents(uint32_t events) override;
+
 private:
   void reEnableEventBasedOnIOResult(const Api::IoCallUint64Result& result, uint32_t event);
+
+  // For windows mimic MSG_PEEK
+  std::unique_ptr<Buffer::Instance> peek_buffer_;
+
+  Api::IoCallUint64Result drainToPeekBuffer();
+  Api::IoCallUint64Result readFromPeekBuffer(void* buffer, size_t length);
+  Api::IoCallUint64Result readFromPeekBuffer(Buffer::Instance& buffer, size_t length);
+  Api::IoCallUint64Result readvFromPeekBuffer(uint64_t max_length, Buffer::RawSlice* slices,
+                                              uint64_t num_slice);
+  Api::IoCallUint64Result peekFromPeekBuffer(void* buffer, size_t length);
+
+  Api::IoCallUint64Result readvImpl(uint64_t max_length, Buffer::RawSlice* slices,
```

Comment:
This is unused.

Reply:
fixed in 1534dcd520695bcb1e871799461681d9d73f0f22